Grammar Updates to the General Folder (#1026)
AdguardTeam/AdguardKnowledgeBase: 01.general/07.how-ad-blocking-works/docs.en.md

To better understand ad blocking, it's important to know the underlying principles of how filters work.

Filtering rules, which filters consist of, aren't created automatically. They are a product of filter maintainers' work, both as professionals and as volunteers. They use browsers' developer consoles and other tools (like AdGuard's filtering log) to determine what rules would block a particular ad or tracker. This is a very simplified description of the process, as some ads are especially hard to block and require multiple rules, multiple iterations, and the use of complex syntax.

> And even when a rule finally gets added to the filter, it doesn't mean that it stays there forever. Ads change, ways to serve the same ads on the same websites change, and so filter rules have to change too. Sometimes rules get obsolete, sometimes a new ad appears, and sometimes a new filtering rule is needed to block the same ad. Filters are often maintained by a single person, but even for a team of maintainers, it's impossible to constantly monitor the entire web. This is why many ad blockers have tools to help users easily report any filter-related issues they encounter.

Author: ```suggestion
And even when a rule finally gets added to a filter, it doesn't mean that it stays there forever. Ads change, ways to serve the same ads on the same websites change, and the filter rules have to change, too. Sometimes rules become obsolete, a new ad appears, or a new filtering rule is needed to block the same ad. Filters are often maintained by one person, but even for a team of maintainers, it's impossible to constantly monitor the entire web. That's why many ad blockers have tools to help users easily report any filter-related issues they encounter.
```

## Types of filtering rules

There are many types of filtering rules that serve different purposes. Depending on the ad blocker you use, and especially on your OS, certain types of rules may not be supported.

<a name="basic"></a>

### Basic filtering rules

To be displayed on a web page or in an app, the ad has to be loaded from a server first. To do so, the browser or the app needs to send a web request. The most basic way of preventing an ad from appearing on your screen is to block this request so it never reaches the server, and thus there's no reply.

Basically, all AdGuard ad blocker products can block web requests according to the active filter rules. This method is very effective at stopping the ad, but it has some drawbacks. The most obvious one is that whatever place the ad was taking up will be left empty or occupied by an ad leftover.

<a name="cosmetic"></a>

### Cosmetic filtering rules

> Every web page has a DOM ("document object model"), which is basically an HTML document that contains the page's structure and all its elements. Ads on the page are also elements and therefore reflected in the DOM. Ad blockers can remove parts of the DOM, and filtering rules help them understand which parts are ads and should be removed, and which shouldn't be messed with.

Author: ```suggestion
Every web page has a Document Object Model (DOM), an HTML document containing the structure and elements of this page. As ads are also page elements, they get recorded in the DOM. Ad blockers can remove parts of the DOM, while filtering rules help them understand which parts are ads and should be removed, and which parts should be left intact.
```
